### PR TITLE
Fix vulnerabilities and tests, upgrade packages and CLI version (AST-38513)

### PR DESCRIPTION
**Pull request by checkmarx-kobi-hagmi**
### Description

> Fix vulnerabilities and tests, upgrade packages and CLI to version 2.1.0

### References

> https://checkmarx.atlassian.net/browse/AST-38513

### Testing

> Manually tested and verified

### Checklist

- [ ] I have added documentation for new/changed functionality in this PR (if applicable).
- [ ] All active GitHub checks for tests, formatting, and security are passing
- [ ] The correct base branch is being used

### CODE REVIEW

**Review by AlvoBen — APPROVED** [review]

File: ast-visual-studio-extension-tests/cx-unit-tests/cx-wrapper-tests/ResultTest.cs [hunk 1]
```diff
@@ -49,7 +49,7 @@ public void TestResultsSummaryJSON()
         [Fact]
         public void TestResultsStructure()
         {
-            List<Scan> scanList = cxWrapper.GetScans();
+            List<Scan> scanList = cxWrapper.GetScans("statuses=Completed");
```
Comment (OrShamirCM) [inline ast-visual-studio-extension-tests/cx-unit-tests/cx-wrapper-tests/ResultTest.cs:52]: why did you add it here?

**Review by OrShamirCM — APPROVED** [review]
